### PR TITLE
Client - copy token to clipboard

### PR DESCRIPTION
**Pull request by yuval-hazaz**
based on PR #2564 by @molaycule 

## PR Details

copy token to clipboard

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Settings/ApiTokenList.tsx [hunk 1]
```diff
@@ -33,10 +34,19 @@ export const ApiTokenList = React.memo(() => {
...
       setNewToken(token);
       setNewTokenState(false);
+      setTokenCopied(false);
     },
     [setNewToken, setNewTokenState]
   );
 
+  const handleCopyToken = async () => {
+    const token = data?.userApiTokens[0].token;
+    if (token) {
```
Comment (tupe12334) [inline packages/amplication-client/src/Settings/ApiTokenList.tsx:44]: For my knowledge of the principles of clean code, you better check if the token doesn't exist and if not, return this function of log some error, not wrap all the code in an if statement

**Comment by tupe12334** [comment]
@yuval-hazaz and @molaycule Why do that a one-time click and not a clickable button next to the new token?
Except that the pr looks good, just look at my review @molaycule nice work!

**Comment by MoFoGo** [comment]
@yuval-hazaz - does this require inclusion in Docs?

File: packages/amplication-client/src/Settings/ApiTokenList.tsx (hunk 1 above)
Comment (tupe12334) [reply packages/amplication-client/src/Settings/ApiTokenList.tsx:44]: @yuval-hazaz do you want to what to this or merge the pr without that?

Comment (molaycule) [reply packages/amplication-client/src/Settings/ApiTokenList.tsx:44]: Hi @tupe12334 do you mean something like this:
```
const token = data?.userApiTokens[0]?.token;
if (!token) return console.log("Token does not exist");
await navigator.clipboard.writeText(token);
setTokenCopied(true);
```
Kindly put me through if I am wrong

**Comment by molaycule** [comment]
> @yuval-hazaz and @molaycule Why do that a one-time click and not a clickable button next to the new token? Except that the pr looks good, just look at my review @molaycule nice work!

Hi @tupe12334, in my opinion having a button informs the user of the copy token to clipboard functionality.

File: packages/amplication-client/src/Settings/ApiTokenList.tsx (hunk 1 above)
Comment (tupe12334) [reply packages/amplication-client/src/Settings/ApiTokenList.tsx:44]:
```const token = data?.userApiTokens[0]?.token;
if (!token) {
console.log("Token does not exist");
return
}
await navigator.clipboard.writeText(token);
setTokenCopied(true);```

Comment (tupe12334) [reply packages/amplication-client/src/Settings/ApiTokenList.tsx:44]: Hi @molaycule more like this, it let you encapsulate the "unhappy" path and continue on the "happy" path

Comment (molaycule) [reply packages/amplication-client/src/Settings/ApiTokenList.tsx:44]: Hi @tupe12334 thank you for the feedbacks, I appreciate the review. Kindly check the PR #2604 to resolve this conversation.

**Review by tupe12334 — APPROVED** [review]

**Comment by tupe12334** [comment]
@mshidlov Please merge this pr or let me know if I can merge it, checked by me and @yuval-hazaz

**Pull request merged**